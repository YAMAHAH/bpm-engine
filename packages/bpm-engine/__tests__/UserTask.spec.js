import fs from 'fs';

import BPMEngine from 'bpm-engine';
import History from './Plugins/History';

describe('UserTask', () => {
  it('Pauses a token and controls continuing it', async (done) => {
    const history = new History();
    const bpm = new BPMEngine({
      plugins: [history],
    });

    // start the process and the flow
    const token = await bpm.createProcessInstance({
      workflowDefinition: fs.readFileSync(`${__dirname}/diagrams/UserTask.bpmn`, 'utf-8'),
    });
    await token.exec();

    // continue the flow
    const continueToken = await bpm.continueTokenInstance({
      tokenId: token.tokenId,
    });
    await continueToken.exec();

    setTimeout(() => {
      expect(history.store).toMatchSnapshot();
      done();
    }, 100);
  });
});
