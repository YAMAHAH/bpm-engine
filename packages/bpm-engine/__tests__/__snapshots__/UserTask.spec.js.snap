// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTask Pauses a token and controls continuing it 1`] = `
Array [
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "StartEvent_1",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_00jvsas",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_00jvsas",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_00jvsas",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "Task_123acd6",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "Task_123acd6",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "Task_123acd6",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "SequenceFlow_0xem89z",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "SequenceFlow_0xem89z",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "SequenceFlow_0xem89z",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
  Object {
    "elementId": "EndEvent_1dwx5c0",
    "lanes": undefined,
    "participant": undefined,
    "state": "ready",
  },
  Object {
    "elementId": "EndEvent_1dwx5c0",
    "lanes": undefined,
    "participant": undefined,
    "state": "active",
  },
  Object {
    "elementId": "EndEvent_1dwx5c0",
    "lanes": undefined,
    "participant": undefined,
    "state": "complete",
  },
]
`;
